File: backend/app/models/activity_log.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.sql import func

from app.core.database import Base


class ActivityLog(Base):
    __tablename__ = "ActivityLogs"

    ActivityID = Column(Integer, primary_key=True, index=True)

    UserID = Column(Integer, ForeignKey("Users.UserID"))

    Action = Column(String(50), nullable=False)

    TableName = Column(String(100), nullable=False)

    RecordID = Column(Integer, nullable=False)

    ActionTime = Column(DateTime(timezone=True), server_default=func.now())

File: frontend/types/activitylog.ts
export interface ActivityLog { ActivityID:number; UserID?:number|null; Action:string; TableName?:string|null; RecordID?:number|null; ActionTime?:string|null; Details?:string|null }
export interface ActivityLogCreate { UserID?:number|null; Action:string; TableName?:string|null; RecordID?:number|null; Details?:string|null }
export type ActivityLogUpdate = Partial<ActivityLogCreate>;
